### PR TITLE
fix: csrf check failed on public share with password

### LINKED ISSUE #44081: [Bug]: CSRF check failed on public share with password
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Error message "CSRF check failed" is shown up after submitted password input.

### Steps to reproduce

1. Open public share link on a tab
2. Open nextcloud login another tab then login success
3. Active the first tab then input password and submit

### Expected behavior

Error message "CSRF check failed" should not be shown up.

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by luka-nextcloud**
* Resolves: #44081

## Summary
"CSRF check failed" on public share with password

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: core/js/publicshareauth.js [hunk 1]
```diff
@@ -52,3 +52,18 @@ document.addEventListener('DOMContentLoaded', function() {
...
+		form.addEventListener('submit', async function(event) {
+			event.preventDefault();
+			var requestToken = document.getElementById('requesttoken');
+			if (requestToken) {
+				requestToken.value = await OC.fetchRequestToken();
+			}
+			form.submit();
+		});
+	}
+});
```
Comment (ChristophWurst) [inline core/js/publicshareauth.js:69]: would it be possible to move this into a "modern" js module that goes through webpack? then you can import `@nextcloud/router` directly and we avoid adding a new property to the dated `OC` global

File: core/src/main.js [hunk 2]
```diff
@@ -50,3 +52,20 @@ window.addEventListener('DOMContentLoaded', function() {
...
+	const form = document.getElementById('password-input-form')
+	if (form) {
+		form.addEventListener('submit', async function(event) {
+			event.preventDefault()
+			const requestToken = document.getElementById('requesttoken')
+			if (requestToken) {
+				const url = generateUrl('/csrftoken')
+				const resp = await Axios.get(url)
+				requestToken.value = resp.data.token
+			}
```
Comment (ChristophWurst) [inline core/src/main.js:67]: Could use grab the currently known CSRF token from `@nextcloud/auth` to avoid the additional request?

Comment (luka-nextcloud) [reply core/src/main.js:67]: This PR is for solving issue that the currently known CSRF token might not be the latest.

**Review by emoral435 — APPROVED** [review]
This looks like a valid solution to me - however, haven't tested it fully

File: core/src/main.js (hunk 2 above)
Comment (ChristophWurst) [reply core/src/main.js:67]: There is an existing CSRF token update mechanism that pulls a fresh token every 30 seconds. Is that not sufficient?

Comment (luka-nextcloud) [reply core/src/main.js:67]: I think it is not sufficient. User might submit form during the gap time and see the CSRF failed error randomly.

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**